rfd/xxxx-bot-for-cert-renewals.md

```diff
@@ -0,0 +1,298 @@
+[CA rotations](https://goteleport.com/docs/architecture/authentication/#certificate-rotation)
+as the renewal has to be performed within the rotation's grace period in order
+to maintain connectivity.
+
+This experience can incentivize users to make decisions that conflict with
+security best practices, including:
+
+- Requesting certificates with a very large time-to-live (TTL)
+- Electing not to perform CA rotations
+
+Both of these decisions are simply delaying the inevitable - eventually, the
+certificates will need to be renewed and external systems will need to be
+updated. The agent proposed in this RFD intends to automate the process of
+renewing these certificates - requiring less manual work on behalf of our users,
+and enabling them to use shorter TTLs and perform frequent CA rotations.
+
+## Details
+
+### Goals
+
+The primary user this agent is designed for is the admin who wants to leverage
+Teleport for as many systems as possible, but simply does not have the time to
+keep track of all certificates and manually renew them.
+
+While users with sophisticated automation and dedicated credential management
+tools may benefit from this solution, they likely already have the resources to 
+automate certificate renewals.
+
+As a result, a primary goal for this tool is simplicity. We will optimize the user
+experience for those looking to add systems to their cluster quickly and easily rather
+than supporting a multitude of configuration options and hooks for customization.
+
+### Overview
+
+We will develop a new lightweight tool, `tbot`, which runs as an agent on
+external systems that are part of a Teleport cluster (a VM running a Postgres
+database, or an OpenSSH server, for example).
+
+This agent will join the Teleport cluster, request certificates, store them in a
+user-specified [destination](#destination), and then monitor the certificates
+and renew them as necessary.
+
+#### User Experience
+
+First, a token will be generated for the bot to join the cluster.
+
+```
+$ tctl tokens add --type=bot
+The invite token: 13b74f49d27536dd5c514073097c197b
+This token will expire in 60 minutes
+
+...
+```
+
+Next, the agent will be started using the generated token (full set of options
+omitted here for brevity):
+
+```
+$ tbot start --name=foo --auth-server=proxy.example.com:3080 --token=TOKEN ...
+```
+
+Note that both direct dialing or connecting through a Teleport proxy are supported.
+
+The bot joins the cluster and can be seen with the `tctl` command:
+
+```
+$ tctl bots ls
+ID              NAME
+0123-4567       foo
+```
+
+TODO: what else might we want to see? public addr? number (and kind) of certs?
+
+In the event of an incident, the bot can be locked with the new `tctl lock`
+functionality. A locked bot will be unable to renew certificates and will have
+its existing certificates revoked. (TODO: this is how it works, right?)
+
+Note that a bot can manage multiple sets of certificates, and locking a
```

Comment:
Please add a separate section on attack vector tree and possible compromise scenarios. E.g. machine with certbot was taken over. How would `lock` mitigate that and what is the blast radius? I assume with impersonation, the impersonator user will be blocked from issuing new certificates and the existing certificates will expire within 5 minute window. Any other possible attack vectors?